sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -562,6 +564,25 @@ private bool TryLoadConstructorArguments(AttributeData attributeData, IDictionar
                 return true;
             }
 
+            private void OverrideBindingName(INamedTypeSymbol attributeClass, ref string argumentName)
+            {
+                var bindingPropertyNameSymbol = Compilation.GetTypeByMetadataName(Constants.BindingPropertyNameAttributeType);
+
+                foreach (var prop in attributeClass.GetMembers().Where(a => a is IPropertySymbol))
+                {
+                    if (String.Equals(prop.Name, argumentName, StringComparison.OrdinalIgnoreCase)) // relies on convention where constructor parameter names match the property their value will be assigned to (JSON serialization is a precedence for this convention)
+                    {
+                        var bindingNameAttrList = prop.GetAttributes().Where(attr => SymbolEqualityComparer.Default.Equals(attr.AttributeClass, bindingPropertyNameSymbol));
+
+                        if (bindingNameAttrList.SingleOrDefault() is { } bindingNameAttr)
+                        {
+                            bindingNameAttr = bindingNameAttrList.FirstOrDefault(); // there will only be one BindingAttributeName attribute b/c there can't be duplicate attributes on a piece of syntax
```

Comment:
Do we need this line? `bindingNameAttr` is already the single item from line 577. So 580 should just work.

Reply:
oh good catch let me update